app/api/dao/user.py

```diff
@@ -9,6 +9,36 @@
 from app.utils.validation_utils import is_email_valid
 
 
+def check_security(password_check):
+    """Returns a boolean if the password is secure"""
+    flag=0
+    if len(password_check) < 8:
+        flag=0
+        return messages.INSUFFICIENT_LENGTH,400
+    else:
+        for ch in password_check:
+            if ch.isupper():
+                flag=1
+                break
+        if flag!=1:
+            return messages.ATLEAST_ONE_UPPERCASE,400
+        for ch in password_check:
+            if ch.isdigit():
+                flag=2
+                break
+        if flag!=2:
+            return messages.ATLEAST_NUMBER,400
+        regex=re.compile('[{(~!@#$%^&*_-|\:;?/+=)}]')
+        if(regex.search(password_check) is None):
+            return messages.ATLEAST_ONE_SPECIAL_CHARACTER,400
+        else:
+            flag=3
+    if flag==3:
+        return True
+    else:
+        return False
+
+
```

Comment:
Hey.
Two things here:
1. You need to add spaces between variables. Basically run your code peace via some python linter/formatter.
2. Can we come up with a more efficient way of this algorithm rather than running the loop multiple times? I was thinking of using a flags dict and updating the count there for each type of flag. That way we will run the loop only twice. Once on the password text and once on the flag dict.
    Using a list of flags/bitwise operation can also be another way.
WDYT?

Reply:
Hi @sara-02
I am really sorry for the late reply.Actually I was down with fever and wasn't able to work since few days. I will surely format the code by adding the spaces.
Using dictionary is a good idea.According to me,there will only be one loop in that case changing the flag according to the password text.But we need to put a break like here so that all letters of the password are not checked once there is one mistake. Shall I modify the code using dict?

Reply:
Hey, we can put a break, but in the current implementation, we are scanning the string again and again for each flag. That is why I thought if we can reduce the number of loops there. Also, what if tomorrow we add a new flag, can our code to flexible to accommodate those changes? A dict of flags or even a bitwise operation on a flag(assuming eight types of flags at a time) might be easier to handle than adding/deleting loops for each change.

Reply:
Let me know if you have any more ideas.

Reply:
What will the eight different types of flags include?

Reply:
I was just floating an idea, the flags will include the same current set of caps, number, special char. I know it is not 8, but giving an idea that we can have bitwise flags as well.

Reply:
Okay I will do the same

Reply:
@sara-02 I've commited the changes you mentioned.Could you please review them?